stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/view/internal/MessageReplyView.kt

```diff
@@ -99,35 +100,48 @@ internal class MessageReplyView : FrameLayout {
         }
     }
 
-    private fun setReplyBackground(message: Message, isMine: Boolean) {
+    private fun setReplyBackground(message: Message, isMine: Boolean, style: MessageReplyStyle?) {
         val shapeAppearanceModel = ShapeAppearanceModel.builder()
             .setAllCornerSizes(REPLY_CORNER_RADIUS)
             .setBottomLeftCornerSize(if (isMine.not()) 0f else REPLY_CORNER_RADIUS)
             .setBottomRightCornerSize(if (isMine) 0f else REPLY_CORNER_RADIUS)
             .build()
-        val isLink = message.attachments
-            .lastOrNull()
-            ?.type == ModelType.attach_link
         binding.replyContainer.background = MaterialShapeDrawable(shapeAppearanceModel).apply {
             when {
-                isLink -> {
+                isLink(message) -> {
                     paintStyle = Paint.Style.FILL
-                    setTint(context.getColorCompat(R.color.stream_ui_blue_alice))
+                    val color = if (isMine) {
+                        style?.linkBackgroundColorMine ?: context.getColorCompat(R.color.stream_ui_blue_alice)
+                    } else {
+                        style?.linkBackgroundColorTheirs ?: context.getColorCompat(R.color.stream_ui_blue_alice)
+                    }
+                    setTint(color)
                 }
                 isMine -> {
-                    paintStyle = Paint.Style.FILL
-                    setTint(context.getColorCompat(R.color.stream_ui_grey_whisper))
+                    paintStyle = Paint.Style.FILL_AND_STROKE
+                    val color =
+                        style?.messageBackgroundColorMine ?: context.getColorCompat(R.color.stream_ui_grey_whisper)
+                    setTint(color)
+                    style?.messageStrokeColorMine?.let {
+                        setStrokeTint(it)
+                    }
+                    strokeWidth = style?.messageStrokeWidthMine ?: DEFAULT_STROKE_WIDTH
                 }
                 else -> {
                     paintStyle = Paint.Style.FILL_AND_STROKE
-                    setStrokeTint(context.getColorCompat(R.color.stream_ui_grey_whisper))
-                    strokeWidth = DEFAULT_STROKE_WIDTH
-                    setTint(context.getColorCompat(R.color.stream_ui_white))
+                    setStrokeTint(style?.messageStrokeColorTheirs ?: context.getColorCompat(R.color.stream_ui_grey_whisper))
+                    strokeWidth = style?.messageStrokeWidthTheirs ?: DEFAULT_STROKE_WIDTH
+                    val tintColor = style?.messageBackgroundColorTheirs ?: context.getColorCompat(R.color.stream_ui_white)
+                    setTint(tintColor)
                 }
             }
         }
     }
 
+    private fun isLink(message: Message) = message.attachments
+        .lastOrNull()
+        ?.type == ModelType.attach_link
```

Comment:
Why a message is a link only when the last attachment is of type link? If the first is a link, then the message is not a link?

Reply:
Fixed